napari/layers/labels/labels.py

```diff
@@ -387,20 +387,22 @@ def color(self, color):
         self.color_mode = color_mode
 
     def _ensure_int_labels(self, data):
-        """Ensure data is integer by converting from bool if required, raising an error otherwise"""
+        """Ensure data is integer by converting from bool if required, raising an error otherwise."""
         looks_multiscale, data = guess_multiscale(data)
         if not looks_multiscale:
             data = [data]
         int_data = []
-        for d in data:
-            if np.issubdtype(d.dtype, np.floating):
+        for data_level in data:
+            if np.issubdtype(data_level.dtype, np.floating):
                 raise TypeError(
-                    f"Only integer types are supported for Labels layers, but data contains {d.dtype}."
+                    trans._(
+                        "Only integer types are supported for Labels layers, but data contains {}."
```

Comment:
Leaving a `{}` is very tempting but I suggest we use a very explanatory variable name so that translators have a clear picture of what {} means.

Reply:
Oh, thanks for pointing that out - I didn't realise the translator would translate the keyword arg names

Reply:
Well, they will not translate the keywords, anything in `{..}` means do not translate, but knowing what it will contains does help a lot when translating.